src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBuffer.cs

```diff
@@ -1109,7 +1109,11 @@ internal void SetToDateTimeOffset(DateTimeOffset dateTimeOffset, byte scale)
             _isNull = false;
         }
 
+#if netcoreapp
+        private static void FillInTimeInfo(ref TimeInfo timeInfo, ReadOnlySpan<byte> timeBytes, int length, byte scale)
```

Comment:
why cant we just keep the span version ?

Reply:
Only the netcore build has a dependency on System.Memory which is required for span. Rather than force all netfx and netcore consumers to pull in that and possibly a full set of interdependent assemblies supporting it I talked with @karelz in gitter and he suggested making it netcore specific. If someone wants to make the decision to require span for all versions then everyone can use the span implementation.

Reply:
The netfx wont use this implementation. AFAIK the implementatation in the src folder is only built and compiled for netcoreapp which makes the if def redundant.

@danmosemsft  am i missing something here ?

Reply:
@stephentoub thoughts about this pattern?
We try to avoid `#if platform` in our implementation.

Reply:
I'm definitely missing something here. The conditions for inclusion of SqlBuffer and TdsStateParserObject in the project file are simply `
Condition="'$(IsPartialFacadeAssembly)' != 'true' AND '$(OSGroup)' != 'AnyOS'"` You assert that netfx won't use this, how do you know this and where can I determine this? If this is the case then surely it still needs to build with `build -allconfigurations` because, again in the original PR I referenced, it didn't and I was asked to change it to pass the CI. 

If neither code level #define's or conditionally included *.netcoreapp.cs files are appropriate to separate out particular functionality just how should it be done? How should this change be made without forcing a dependency on System.Memory on all consumers?

Reply:
This source code is going to be compiled against netstandard, netcoreapp, uap and netfx, however for netfx this will be a partial facade assembly and none of the source files are included for netfx. We know this because `IsPartialFacadeAssembly` is set to true when targeting netfx: https://github.com/dotnet/corefx/blob/c26eb56a6222a581fb51b672c6503ba2ead3fdca/src/System.Data.SqlClient/src/System.Data.SqlClient.csproj#L6

And System.Memory is inbox in Uap and Netcoreapp so there shouldn't be a problem on adding a reference to it:

https://github.com/dotnet/corefx/blob/c26eb56a6222a581fb51b672c6503ba2ead3fdca/src/System.Memory/Directory.Build.props#L9-L10

System.Memory is already referenced in this project when TargetsWindows == true, so it won't hurt moving this to when `IsPartialFacade != true and AnyOS != true`. For netstandard it doesn't hurt anything to add this dependency, as System.Data.SqlClient in netstandard is an OOB package.

Reply:
> This source code is going to be compiled against netstandard, netcoreapp, uap and netfx, however for netfx this will be a partial facade assembly and none of the source files are included for netfx.

Agreed.

> And System.Memory is inbox in Uap and Netcoreapp so there shouldn't be a problem on adding a reference to it

That is correct.